rs/replicated_state/src/canister_state/queues.rs

```diff
@@ -1008,7 +1098,20 @@ impl CanisterQueues {
     ) {
         use Context::*;
 
+        // If this is an inbound response, remember its `originator_reply_callback`, so
+        // we can later produce an `UnknownResponse` for it, when popped.
         let context = reference.context();
+        if let (Inbound, RequestOrResponse::Response(response)) = (context, msg) {
+            assert_eq!(
+                None,
+                self.shed_responses
+                    .insert(reference, response.originator_reply_callback)
+            );
+
+            // Leave the input queue unchanged, as "dropped responses" are non-stale.
```

Comment:
We can return early here because the stats related to the dropped response are tracked in the memory pool and it won't affect the queue stats, right?

Reply:
Correct.

And because we're not actually dropping the response, merely replacing what it points to, this has no effects on the number of slots / reservations maintained in `QueueStats`.